src/System.Data.SqlClient/src/System/Data/Locale/Locale.cs

```diff
@@ -32,7 +32,7 @@ static Locale()
             {
                 throw ADP.ArgumentOutOfRange("lcid");
             }
-            return s_cachedEncodings.GetOrAdd(lcid, GetDetailsInternal);
+            return s_cachedEncodings.GetOrAdd(lcid, id => GetDetailsInternal(id));
```

Comment:
Put a comment here to nobody "fixes it". TBH, the compiler should just fix this bug. @stephentoub reported it like 2 years ago... 

/cc @jaredpar